src/object/obj_rpc.c

```diff
@@ -138,19 +138,19 @@ crt_proc_daos_iod_and_csum(crt_proc_t proc, crt_proc_op_t proc_op,
 
 	rc = crt_proc_d_iov_t(proc, &iod->iod_name);
 	if (unlikely(rc))
-		return rc;
+		D_GOTO(out, rc);
```

Comment:
We've been moving away from D_GOTO() so I don't think this part of the PR adds much although we don't have a consistent approach to D_GOTO/goto/return.

Reply:
I tried to use the same style as in the rest of function. If is use just "goto" I use it instead of macro. We should have a strict rule about this. I'm not aware about this yet.